internal/cli/images.go

```diff
@@ -41,11 +46,18 @@ func runImagesCmd(cmd *cobra.Command, args []string) error {
 			return err
 		}
 
-		err = file.WriteFile(bb, fmt.Sprintf("images/%d.jpg", i))
+		err = file.WriteFile(bb, fmt.Sprintf("%s/%d.jpg", output, i))
 		if err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
+
+func checkIfDirectoryExistsAndCreate(folder string) bool {
+	if err := os.Mkdir(folder, 0755); os.IsExist(err) {
+		return true
+	}
+	return false
+}
```

Comment:
The signature is a bit misleading, and also the error is not fully checked, and the boolean return value is not used.

I would use also the `os. MkdirAll` to make it work with subdirectories, and use less open permissions (I don't think we need the images to be executable).

```go
func createOutputDirectory(output string) error {
	if err := os.MkdirAll(output, 0644); !os.IsExist(err) {
		return err
	}
	return nil
}
```

By the way, since they are just three lines and we need to check the error anyway I would avoid putting them in a separate function for the moment.

So I would just use these at the beginning, maybe with a comment

```go
// create the output folder if it doesn't exists
if err := os.MkdirAll(output, 0644); !os.IsExist(err) {
	return err
}
```

Reply:
About the permissions it seems that we need the dir to be executable (see https://github.com/enrichman/stegosecrets/pull/22#discussion_r1003247733). So it's fine to have `0755`.